src/app/components/Navbar.tsx

```diff
@@ -1,45 +1,60 @@
 'use client';
 
-import { Suspense, useState } from 'react';
+import { Fragment, useCallback, useMemo, useState } from 'react';
 
 import Link from 'next/link';
 import { usePathname, useSearchParams } from 'next/navigation';
 
 import BackButton from './BackButton';
 
+import allDuas from 'duas.json';
+
+interface INavComponentProps {
+  navbar?: boolean;
+  toggle: () => void;
+}
+
+const menu = [
+  { name: 'All', url: { pathname: '/Duas' } },
+  { name: 'Quranic Duas', url: { pathname: '/Duas', query: { tag: 'Quran' } } },
+  { name: 'Masnoon Duas', url: { pathname: '/Duas', query: { tag: 'Masnoon' } } },
+  { name: 'Sleeping', url: { pathname: '/Duas', query: { tag: 'Sleep' } } },
+  { name: 'Mosque', url: { pathname: '/Duas', query: { tag: 'Mosque' } } },
+  { name: 'House', url: { pathname: '/Duas', query: { tag: 'House' } } },
+];
+
 const Navbar = () => {
-  const pathname = usePathname();
   const searchParams = useSearchParams();
-
   const [navbar, setNavbar] = useState(false);
-  const menu = [
-    { name: 'All', url: { pathname: '/Duas' } },
-    { name: 'Quranic Duas', url: { pathname: '/Duas', query: { tag: 'Quran' } } },
-    { name: 'Masnoon Duas', url: { pathname: '/Duas', query: { tag: 'Masnoon' } } },
-    { name: 'Sleeping', url: { pathname: '/Duas', query: { tag: 'Sleep' } } },
-    { name: 'Mosque', url: { pathname: '/Duas', query: { tag: 'Mosque' } } },
-    { name: 'House', url: { pathname: '/Duas', query: { tag: 'House' } } },
-  ];
+  const toggleNavbar = useCallback(() => setNavbar(prev => !prev), []);
+
+  const otherTags = useMemo(() => {
+    const mTags = menu.reduce((acc: string[], qParams) => {
+      if (qParams.url.query && qParams.url.query.tag) {
+        acc.push(qParams.url.query.tag);
+      }
+      return acc;
+    }, []);
+
+    const dTags = allDuas.flatMap(d => d.tags);
```

Comment:
Please use the comprehensive and self-evident variable names. 

Rule of thumb: Code is read more than it is executed. 😄